pkg/logs/tailers/windowsevent/tailer.go

```diff
@@ -89,210 +98,243 @@ func (t *Tailer) Identifier() string {
-	}
-	for _, value := range values {
-		valueMap, ok := value.(map[string]interface{})
-		if !ok {
-			continue
-		}
+func retryForeverWithCancel(ctx context.Context, operation backoff.Operation) error {
+	resetBackoff := backoff.NewExponentialBackOff()
+	resetBackoff.InitialInterval = 1 * time.Second
+	resetBackoff.MaxInterval = 1 * time.Minute
+	// retry never stops if MaxElapsedTime == 0
+	resetBackoff.MaxElapsedTime = 0
 
-		// Get element value
-		text, foundText := valueMap["#text"]
-		// Qualifier optional
-		qualifier, foundQualifier := valueMap["Qualifiers"]
-		if foundText && foundQualifier {
-			// Remove Qualifiers attribute from EventID by
-			// overwriting the path with just the text value
-			_ = mv.SetValueForPath(text, eventIDPath)
-			// Add qualifier value to a new path
-			_ = mv.SetValueForPath(qualifier, eventIDQualifierPath)
-		}
-	}
-	return nil
+	return backoff.Retry(operation, backoff.WithContext(resetBackoff, ctx))
 }
 
-// extractDataField transforms the fields parsed from <Data Name='NAME1'>VALUE1</Data><Data Name='NAME2'>VALUE2</Data> to
-// a map that will be JSON serialized to {"NAME1": "VALUE1", "NAME2": "VALUE2"}
-// Data fields always have this schema:
-// https://docs.microsoft.com/en-us/windows/desktop/WES/eventschema-complexdatatype-complextype
-func extractDataField(mv mxj.Map) (map[string]interface{}, error) {
-	values, err := mv.ValuesForPath(dataPath)
-	if err != nil || len(values) == 0 {
-		return nil, fmt.Errorf("could not find path: %s", dataPath)
-	}
-	nameTextMap := make(map[string]interface{})
-	for _, value := range values {
-		valueMap, ok := value.(map[string]interface{})
-		if !ok {
-			continue
+func (t *Tailer) eventLoop(ctx context.Context) {
+	for {
+		// check if loop should exit
+		select {
+		case <-ctx.Done():
+			return
+		default:
 		}
-		name, foundName := valueMap["Name"]
-		text, foundText := valueMap["#text"]
-		if !foundName || !foundText {
-			continue
+
+		// if subscription is not running, try to start it with an exponential backoff
+		if !t.sub.Running() {
+			err := retryForeverWithCancel(ctx, func() error {
+				err := t.sub.Start()
+				if err != nil {
+					t.logErrorAndSetStatus(fmt.Errorf("failed to start subscription: %w", err))
+					return err
+				}
+				// subscription started!
+				return nil
+			})
+			if err != nil {
+				// subscription failed to start, retry returned, probably because
+				// ctx was cancelled. go back to top of loop to check for cancellation
+				// and exit or continue looping as appropriate.
+				continue
+			}
+			// subscription started!
+			t.source.Status.Success()
 		}
-		nameString, ok := name.(string)
-		if !ok {
-			continue
+
+		// subscription is running, wait for and get events
+		select {
+		case <-ctx.Done():
+			return
+		case events, ok := <-t.sub.GetEvents():
+			if !ok {
+				// error
+				err := t.sub.Error()
+				t.logErrorAndSetStatus(fmt.Errorf("GetEvents failed, stopping subscription: %w", err))
+				t.sub.Stop()
```

Comment:
This is interesting and I think right approach (not sure if I thought about it myself). I feel it need a some comment explaining it like ... ok we got and error not sure the state of subscription anymore, let's start from scratch.... (presuming I have captured intent correctly).